src/app/modules/pillbox/pillbox.component.ts

```diff
@@ -0,0 +1,188 @@
+    // for handling various *Content* types
+    if ("numerator" in this.data) {
+      // Handle PillboxFraction - a bespoke usecase
+      let { left, right } = this.makeFractionContent(this.data);
+      this.contentFirst = left;
+      this.contentSecond = right;
+      this.contentFraction = this.data;
+    } else if ("left" in this.data) {
+      // Handle PillboxContentPair
+      // PillboxContent or PillboxColumnContent on either side with a vertical divider
+      let { left, right } = this.makePairContent(this.data);
+      this.contentFirst = left;
+      this.contentSecond = right;
+    } else if ("top" in this.data) {
+      // Handle a single PillboxColumnContent
+      this.setContent(this.data.top);
+      this.setContent(this.data.bottom);
+      this.contentFirst = this.data;
+    } else if ("label" in this.data) {
+      // Handle a single PillboxContent
+      this.setContent(this.data);
+      this.contentFirst = {
+        top: this.data,
+        bottom: null,
+      };
+    }
+  }
+
+  /**
+   * Modifies PillboxContent c to handle customization, or set defaults as required
+   *
+   * @remarks
+   * 'classes' will override default styling. 'bold' always appends.
+   *
+   * @param c PillboxContent
+   */
+  private setContent(c: PillboxContent) {
+    // Set default style/classes if none provided
+    if (!c.classes) {
+      c.classes = 'ds-u-font-size--base'
+    }
+
+    if (c.bold) {
+      c.classes = `${c.classes} ds-u-font-weight--bold`;
+    }
+
+    // Honor provided TooltipModel, otherwise set default with provided tooltip content
+    if (c.tooltip || c.tooltipModel) {
+      c.tooltipModel = c.tooltipModel ?? this.makeTooltipModel(c.tooltip, c.label);
+    }
+
+    if (!c.hqrDataTest) {
+      c.hqrDataTest = '';
+    }
+  }
+
+  /**
+   * Creates tooltip with UX defaults for this component, using only label and tooltip text
+   *
+   * @param tip text label for the tooltip
+   * @param label text label to receive the tooltip
+   * @returns TooltipModel via tooltip.models / app-tooltip
+   */
+  private makeTooltipModel(tip: string, label: string): TooltipModel {
+    let model : TooltipModel;
+
+    model = new TooltipModel({
+      triggerContent: label,
+      tooltipContent: tip,
+      maxWidth: 148,
+      id: '', // otherwise it ends up 'undefined' or 'null' as strings in HTML
+    });
+
+    return model;
+  }
+
+  /**
+   * Creates left and right PillboxColumnContent, with applied defaults or overrides,
+   * allowing passing in a single PillboxContent for either half.
+   *
+   * @param pair PillboxContent or PillboxColumnContent for each side
+   * @returns left and right PillboxColumnContent, as required by the html tempalte
+   */
+  private makePairContent(pair: PillboxContentPair): {
+    left: PillboxColumnContent,
+    right: PillboxColumnContent,
+  } {
+    var left: PillboxColumnContent;
+    var right: PillboxColumnContent;
+
+    if ("top" in pair.left) {
```

Comment:
Should this be in single quotes here and Line 126?

Reply:
oooh, nice catch - thx, change inbound